### DISCUSSION TITLE
How to draw points with vsg::StateInfo?

### DISCUSSION THREAD

**Discussion by redflare2023** [vsg-dev/VulkanSceneGraph#733] [Q&A]
whether can we draw point with vsg::StateInfo?

**Comment by robertosfield** [comment]
vsg::StateInfo if only for passing in state related settings to vsg::Builder::create*(geomInfo, stateInfo)  methods, none of which support points, the available create methods are:

       ref_ptr<Node> createBox(const GeometryInfo& info = {}, const StateInfo& stateInfo = {});
        ref_ptr<Node> createCapsule(const GeometryInfo& info = {}, const StateInfo& stateInfo = {});
        ref_ptr<Node> createCone(const GeometryInfo& info = {}, const StateInfo& stateInfo = {});
        ref_ptr<Node> createCylinder(const GeometryInfo& info = {}, const StateInfo& stateInfo = {});
        ref_ptr<Node> createDisk(const GeometryInfo& info = {}, const StateInfo& stateInfo = {});
        ref_ptr<Node> createQuad(const GeometryInfo& info = {}, const StateInfo& stateInfo = {});
        ref_ptr<Node> createSphere(const GeometryInfo& info = {}, const StateInfo& stateInfo = {});
        ref_ptr<Node> createHeightField(const GeometryInfo& info = {}, const StateInfo& stateInfo = {});

The Builder class is currently written just for creating primitive shapes not general purposes geometry creation.

You will need to create your own geometry.  The vsgPoints library has some examples of point rendering.
